### PR TITLE
RUN-3656: support for resizeSides

### PR DESCRIPTION
**Pull request by wenjunche**
Tests:

[W10](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/5a2963d56f202a5432b79ef7)
[W7](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/5a29634a6f202a5432b79ef6)

[javascript adapter PR](https://github.com/openfin/javascript-adapter/pull/366) 
[Runtime PR](https://github.com/openfin/runtime/pull/764)

### CODE REVIEW

**Comment by datamadic** [comment]
Do the adapters need to also be updated with this new value?

**Comment by HarsimranSingh** [comment]
Can this field be a part of the `resizeRegion` option object? Something like
```js
{
  resizeRegion: {
    size: 20,
    bottomRightCorner: 20,
    sides: 'top,bottom,right,left' // or maybe edges instead of sides?
  }
}
```

**Comment by wenjunche** [comment]
updates to have sides inside resizeRegion

Tests
[W10](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/5a3984f86f202a5432b79f71)
[W7](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/5a39842d6f202a5432b79f70)

File: src/browser/api/window.js [hunk 1]
```diff
@@ -178,6 +178,9 @@ let optionSetters = {
                 });
                 browserWin.setResizeRegion(resizeRegion.size);
                 browserWin.setResizeRegionBottomRight(resizeRegion.bottomRightCorner);
+                if (resizeRegion.sides) {
```
Comment (HarsimranSingh) [inline src/browser/api/window.js]: Don't think there's a need to check if `resizeRegion.sides` is defined. If it is `''`, then it'll do nothing, correct?

Comment (HarsimranSingh) [reply src/browser/api/window.js]: Add the default value for `sides` here: https://github.com/HadoukenIO/core/pull/267/files#diff-d5752a5b2c3b7ce27c77d78f1b9278a6R175

File: src/browser/api/window.js [hunk 2]
```diff
@@ -317,18 +320,23 @@ let optionSetters = {
...
-            return;
-        }
-
-        applyAdditionalOptionsToWindowOnVisible(browserWin, () => {
-            let frame = getOptFromBrowserWin('frame', browserWin, true);
-            if (!frame) {
-                browserWin.setResizeRegion(newVal.size);
-                browserWin.setResizeRegionBottomRight(newVal.bottomRightCorner);
+        if (newVal) {
+            if (typeof newVal.size === 'number' && typeof newVal.cls === 'number') {
```
Comment (HarsimranSingh) [inline src/browser/api/window.js]: What is `newVal.cls`?

File: src/browser/api/window.js [hunk 3]
```diff
@@ -1947,6 +1955,9 @@ function applyAdditionalOptionsToWindow(browserWindow) {
             browserWindow.setResizeRegion(options.resizeRegion.size);
             browserWindow.setResizeRegionBottomRight(options.resizeRegion.bottomRightCorner);
         }
+        if (options.resizeRegion.sides) {
```
Comment (HarsimranSingh) [inline src/browser/api/window.js]: Why isn't this inside the check `if (typeof(newVal) === 'boolean')`

File: src/browser/api/window.js [hunk 4]
```diff
@@ -317,18 +320,23 @@ let optionSetters = {
...
+                    let frame = getOptFromBrowserWin('frame', browserWin, true);
+                    if (!frame) {
+                        browserWin.setResizeRegion(newVal.size);
+                        browserWin.setResizeRegionBottomRight(newVal.bottomRightCorner);
+                    }
+                });
             }
-        });
-        setOptOnBrowserWin('resizeRegion', newVal, browserWin);
+            if (typeof newVal.sides === 'string') {
```
Comment (HarsimranSingh) [inline src/browser/api/window.js]: Could you do something similar for all the fields in the `resizeRegion` object?
```js
if (typeof newVal.size === 'number') {
  applyAdditionalOptionsToWindowOnVisibile(browserWin, () => {...})
}
if (typeof newVal.bottomRightCorner === 'number') {
  applyAdditional...
}
if (typeof newVal.sides === 'string') {
  applyAdditional...
}
```

File: src/browser/api/window.js (hunk 3 above)
Comment (HarsimranSingh) [reply src/browser/api/window.js]: Also no need for the check

**Comment by wenjunche** [comment]
updated with Sim's feedbacks

[W10](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/5a53cc7cbd581266877d664d)
[W7](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/5a53cba9bd581266877d664c)

File: src/browser/api/window.js [hunk 5]
```diff
@@ -1947,6 +1954,7 @@ function applyAdditionalOptionsToWindow(browserWindow) {
             browserWindow.setResizeRegion(options.resizeRegion.size);
             browserWindow.setResizeRegionBottomRight(options.resizeRegion.bottomRightCorner);
         }
+        browserWindow.setResizeSides(options.resizeRegion.sides);
```
Comment (HarsimranSingh) [inline src/browser/api/window.js]: this should be inside `if (!options.frame)`

**Comment by HarsimranSingh** [comment]
Need another approval.

**Comment by HarsimranSingh** [comment]
@wenjunche Add a manual test for this feature

**Comment by wenjunche** [comment]
Tests for new API format

[W10](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/5a5696d4bd581266877d6694)
[W7](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/5a569633bd581266877d6693)

[Manual Test](https://github.com/openfin/test_apps/pull/17)

Updated PRs for Runtime and Javascript adapter

**Pull request merged**